api/app/src/Repository/ReportRepository.php

```diff
@@ -131,8 +131,17 @@ public function getAllByDeterminant(mixed $orgIdsOrUserId, int $determinant, Par
         } else {
             $qb
                 ->select(('count' === $select) ? 'COUNT(DISTINCT r)' : 'r,c,o')
-                ->leftJoin('r.client', 'c')
-                ->leftJoin('c.organisation', 'o')
+                ->innerJoin('r.client', 'c')
+                ->innerJoin('c.organisation', 'o')
+                // join report table to itself to calculate the earliest due date for dual reports if they exist
+                ->innerJoin(
+                    'App\Entity\Report\Report',
+                    'r2',
+                    'WITH',
+                    'r2.client = r.client'
+                )
+                // allows sorting by the earliest due date for each grouped set of reports linked to the same client
+                ->addSelect('MIN(r2.dueDate) AS HIDDEN minDueDate')
                 ->where('o.isActivated = true AND o.id in ('.implode(',', $orgIdsOrUserId).')');
```

Comment:
You need to be careful with implode in SQL. You should use a parameter instead, and pass an array as the value of the parameter, something like:
```
->where('o.isActivated = true AND o.id in :orgIdsOrUserId')
->setParameter('orgIdsOrUserId', $orgIdsOrUserId)
```

You might need to experiment with this a bit, as I'm not sure how Doctrine works with parameter values which are arrays.